pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -201,31 +201,6 @@ func validatePluginConfig(path *field.Path, apiVersion string, profile *config.K
 		"VolumeBinding":               ValidateVolumeBindingArgs,
 	}
 
-	seenPluginConfig := make(sets.String)
-	for i := range profile.PluginConfig {
-		pluginConfigPath := path.Child("pluginConfig").Index(i)
-		name := profile.PluginConfig[i].Name
-		args := profile.PluginConfig[i].Args
-		if seenPluginConfig.Has(name) {
-			errs = append(errs, field.Duplicate(pluginConfigPath, name))
-		} else {
-			seenPluginConfig.Insert(name)
-		}
-		if validateFunc, ok := m[name]; ok {
-			// type mismatch, no need to validate the `args`.
-			if reflect.TypeOf(args) != reflect.ValueOf(validateFunc).Type().In(1) {
-				errs = append(errs, field.Invalid(pluginConfigPath.Child("args"), args, "has to match plugin args"))
-				return errs
```

Comment:
why not just remove this return? We simply shouldn't just return in API validation.

Reply:
because the following couple of lines are calling the validation func to validate the args, will adjust the logic a little bit.

Reply:
but I think it might still better to validate the plugin config after the validation of the removed plugins, what do you think?